engine/grid.py

```diff
@@ -202,178 +222,435 @@ def get_neighbor_node(self, row, col, row_max, col_max):
         else:
             return neighbor_node
 
-
     def activate_rectangle(self, row, col, row_limit, col_limit):
-
         """
         Activates all the nodes in a rectangle based on row start/end, col
         start/end.
         """
         for x in range(row, row_limit):
             for y in range(col, col_limit):
-                self.nodes[x,y].is_active = True
+                self.nodes[x, y].is_active = True
         return self.nodes
 
-
-    
     def activate_circle(self, circle_center_row, circle_center_col, circle_radius):
         """
         Activates all the nodes in a circle based on center and radius.
         """
-        rows = self.nodes.shape[0]
-        cols = self.nodes.shape[0]
-        for x in range(rows):
-            for y in range(cols):
-                if ((x-circle_center_row)**2 + (y-circle_center_col)**2 - circle_radius**2) < 0:
+        for x in range(self.num_rows):
+            for y in range(self.num_cols):
+                if ((x - circle_center_row) ** 2 + (y - circle_center_col) ** 2 - circle_radius ** 2) < 0:
                     self.nodes[x, y].is_active = True
 
-   
-
-    def is_inside_triangle(self, x1, y1, x2, y2, x3, y3, x, y):
-        """
-        A function to check whether point P(x, y) lies inside the triangle 
-        formed by A(x1, y1), B(x2, y2) and C(x3, y3)
-        """
-        def area(x1, y1, x2, y2, x3, y3):
-
-            return abs((x1 * (y2 - y3) + x2 * (y3 - y1)
-                        + x3 * (y1 - y2)) / 2.0)
+    def isInsideTriangle(self, p1, p2, p3, p4):
+        """ A function to check whether point P(x, y) lies inside the triangle formed by
+        A(x1, y1), B(x2, y2) and C(x3, y3)"""
```

Comment:
If we are changing to p1, p2, p3, p4 we should specify the parameters and this precondition in the specification (e.g. p1,p2,... are tuples representing (x,y) points on our grid

Reply:
changed